warp10/src/main/java/io/warp10/continuum/gts/GTSHelper.java

```diff
@@ -107,7 +108,14 @@ public class GTSHelper {
   private static final boolean labelsIdSlowImpl;
 
   static {
-    labelsIdSlowImpl = "true".equals(System.getProperty(LABELSID_SLOWIMPL));
+    if ("17".equals(System.getProperty("java.specification.version")) && null == WarpConfig.getProperty(LABELSID_SLOWIMPL)) {
```

Comment:
Set labelsIdSlowImpl variable first, then test if jdk==17 and labelsIdSlowImpl == false.

Some user may set/keep labelsIdSlowImpl = false in their configuration.

Reply:
That's exactly what the code is doing, it only prints the warning if the labelsid.slowimpl key is not set AND the JDK is 17.

As there is no default value set for labelsid.slowimpl in the configuration the error WILL be displayed when using JDK17 unless the configuration key has been set explicitely.

Reply:
OK, this was intentional.